iOS_SDK/OneSignalSDK/OneSignalCore/Source/API/OneSignalClient.m

```diff
@@ -197,16 +197,21 @@ - (void)handleJSONNSURLResponse:(NSURLResponse*)response data:(NSData*)data erro
     
     NSHTTPURLResponse* HTTPResponse = (NSHTTPURLResponse*)response;
     NSInteger statusCode = [HTTPResponse statusCode];
+    NSDictionary *headers = [HTTPResponse allHeaderFields];
     NSError* jsonError = nil;
     NSMutableDictionary* innerJson;
     
     if (data != nil && [data length] > 0) {
         innerJson = [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingMutableContainers error:&jsonError];
         innerJson[@"httpStatusCode"] = [NSNumber numberWithLong:statusCode];
+        innerJson[@"headers"] = headers;
+        
+        [OneSignalLog onesignalLog:ONE_S_LL_VERBOSE message:[NSString stringWithFormat:@"network request (%@) with URL %@ and headers: %@", NSStringFromClass([request class]), request.urlRequest.URL.absoluteString, request.additionalHeaders]];
```

Comment:
Extraneous log here. `prettyPrintDebugStatementWithRequest` above will log this request, we can have that log the header as well. Note that it does not log GETs, so you can do something similar to https://github.com/OneSignal/OneSignal-iOS-SDK/pull/1473/commits/ac887b9ada7a007af4162d4f6818b40e924d2ad8 to log the GET calls